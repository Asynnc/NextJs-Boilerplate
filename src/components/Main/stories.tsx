import { Story, Meta } from '@storybook/react/types-6-0'
import Main from '.'

export default {
  title: 'Main',
  component: Main,
  args: {
    title: 'Boilerplate',
    description: 'Typescript, ReactJs, NextJs e Styled Components',
  },
} as Meta

export const Default: Story = (args) => <Main {...args} />
export const Basic: Story = (args) => <Main {...args} />
Basic.args = {
  title: 'Example Title',
  description: 'Example Description',
}
