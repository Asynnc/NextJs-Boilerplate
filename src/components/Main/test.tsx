import { render, screen } from '@testing-library/react'

import Main from '.'

describe('<Main/>', () => {
  it('Should be render a heading', () => {
    const { container } = render(<Main />)

    expect(
      screen.getByRole('heading', { name: /Ellie.On Solutions/i })
    ).toBeInTheDocument()

    expect(container.firstChild).toMatchSnapshot()
  })

  it('Should render colors correctly', () => {
    const { container } = render(<Main />)

    expect(container.firstChild).toHaveStyle({ 'background-color': '#06092B' })
  })
})
