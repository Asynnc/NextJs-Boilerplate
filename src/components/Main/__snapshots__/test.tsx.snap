// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Main/> Should be render a heading 1`] = `
<main>
  <h1>
    React Avan√ßado
  </h1>
</main>
`;
